pkgs/games/scummvm/default.nix

```diff
@@ -14,6 +14,14 @@ stdenv.mkDerivation rec {
     hash = "sha256-8/q16MwHhbbmUxiwJOHkjNxrnBB4grMa7qw/n3KLvRc=";
   };
 
+  patches = [
+    # Fix building with Freetype 2.13.3. Remove after next release.
+    (fetchpatch {
+      url = "https://github.com/scummvm/scummvm/commit/65977961b20ba97b1213b5267da0cb1efb49063b.patch";
```

Comment:
The repository is big enough that GitHub has expanded the hashes used in the `index …` line to fit. It will likely grow again, leading to reproducibility issues with the patches (as unfortunately neither `fetchpatch` nor the newer `fetchpatch2` handle stripping out this information yet). Could you append `?full_index=1` to the URL and recalculate the hash? Otherwise LGTM, thank you!

Reply:
Interesting! I've just pushed that change up.

Reply:
I am seeing something a bit odd - I get different a different hash when I omit `hash =` (same as without `full_index=1`: `sha256-e5dJd3gP8OAD3hEJlvOhMemsNErCKTn7avlprApFig0=`) than when I do this:

```
$ nix-build -E '(import <nixpkgs> {}).fetchurl { url = "https://github.com/scummvm/scummvm/commit/65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1"; }'
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
this derivation will be built:
  /nix/store/mgfsmvmk1g16yl6fi895p93vqc6a4z69-65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1.drv
building '/nix/store/mgfsmvmk1g16yl6fi895p93vqc6a4z69-65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1.drv'...

trying https://github.com/scummvm/scummvm/commit/65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2470  100  2470    0     0  12042      0 --:--:-- --:--:-- --:--:-- 12167
error: hash mismatch in fixed-output derivation '/nix/store/mgfsmvmk1g16yl6fi895p93vqc6a4z69-65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-Lu+t7xCvRyn2kSzigdqv4S6rzlLX+LYvwj3jDudo65s=
```

Reply:
Actually I am seeing something a bit odd going on. The hash didn't change (the one I've just pushed was from fetchurl which seems to be different).

Reply:
I think `fetchpatch` is actually normalizing the patch to omit the commit hashes in this case:
```text
$ nix-build -E '(import <nixpkgs> {}).fetchpatch { url = "https://github.com/scummvm/scummvm/commit/65977961b20ba97b1213b5267da0cb1efb49063b.patch"; hash = "sha256-e5dJd3gP8OAD3hEJlvOhMemsNErCKTn7avlprApFig0="; }'
/nix/store/kgvdq0j6295rqm6c9wdd6cspbqdjhjnn-65977961b20ba97b1213b5267da0cb1efb49063b.patch
$ head -n5 /nix/store/kgvdq0j6295rqm6c9wdd6cspbqdjhjnn-65977961b20ba97b1213b5267da0cb1efb49063b.patch
--- a/engines/ags/lib/freetype-2.1.3/autohint/ahglyph.cpp
+++ b/engines/ags/lib/freetype-2.1.3/autohint/ahglyph.cpp
@@ -296,7 +296,11 @@
        AH_Point point = outline->points;
        AH_Point point_limit = point + outline->num_points;
```

Reply:
Oh, wow, it’s even more cursed than I thought… `fetchpatch` does this correctly, but the new and improved(!) `fetchpatch2` *doesn’t*. Okay, this is fine then, sorry for the confusion!

Reply:
Yeah it's not great. Seems like it's known: https://github.com/NixOS/nixpkgs/issues/257446